src/server/game/Entities/Creature/Creature.cpp

```diff
@@ -3245,8 +3245,12 @@ void Creature::ClearTextRepeatGroup(uint8 textGroup)
 
 bool Creature::CanGiveExperience() const
 {
-    return !IsCritter()
-        && !IsPet()
+    CreatureTypeEntry const* creatureType = sCreatureTypeStore.AssertEntry(GetCreatureType());
+
+    if (creatureType->HasFlag(CreatureTypeEntryFlags::CREATURE_TYPE_ENTRY_FLAGS_NO_EXPERIENCE))
```

Comment:
You should revert this change, this flag has a different meaning, its a "do not include in tab targeting"

Reply:
Ohhh ok, that actually makes a lot of sense now. The information I was basing it on claimed it was something else.

Reply:
Partially reverted that commit now.